src/extension/devtools/devtools.ts

```diff
@@ -48,94 +49,93 @@ devtools.addConnection(EXPLORER_SUBSCRIPTION_TERMINATION, () => {
   sendMessageToClient(EXPLORER_SUBSCRIPTION_TERMINATION);
 });
 
-devtools.listen(CREATE_DEVTOOLS_PANEL, ({ payload }) => {
+devtools.listen(CREATE_DEVTOOLS_PANEL, async ({ payload }) => {
   if (!isPanelCreated) {
-    chrome.devtools.panels.create(
+    const panel = await browser.devtools.panels.create(
       "Apollo",
       "logo_devtools.png",
       "panel.html",
-      function (panel) {
-        isPanelCreated = true;
-        const { queries, mutations, cache } = JSON.parse(payload);
-        let removeUpdateListener;
-        let removeExplorerForward;
-        let removeSubscriptionTerminationListener;
-        let removeReloadListener;
-        let clearRequestInterval;
-        let removeExplorerListener;
-
-        panel.onShown.addListener((window) => {
-          sendMessageToClient(PANEL_OPEN);
-
-          const {
-            __DEVTOOLS_APPLICATION__: {
-              initialize,
-              writeData,
-              receiveExplorerRequests,
-              receiveSubscriptionTerminationRequest,
-              sendResponseToExplorer,
-              handleReload,
-              handleReloadComplete,
-            },
-          } = window as any;
-
-          if (!isAppInitialized) {
-            initialize();
-            writeData({ queries, mutations, cache: JSON.stringify(cache) });
-            isAppInitialized = true;
-          }
+    );
 
-          clearRequestInterval = startRequestInterval();
+    isPanelCreated = true;
```

Comment:
I would even move the `isPanelCreated = true;` before the asynchronous call to `browser.devtools.panels.create` so we don't get into any race condition where this can be called multiple times while the panel is being created, resulting in multiple panels (probably impossible or it would already have happened before, but better be sure).

Reply:
I might leave this for now only because its worked like this for a while. I would absolutely love to come back to some of this logic anyways and perhaps we could do a refactor of this code a bit more. Totally agree though that its possible this could create race conditions.

Perhaps we could put this in a separate PR that calls this out a bit better?